var/spack/repos/builtin/packages/shiny-server/package.py

```diff
@@ -44,24 +44,29 @@ class ShinyServer(Package):
 
     version('1.5.3.838', '96f20fdcdd94c9e9bb851baccb82b97f')
 
-    depends_on('python@2.9.99')  # docs say: "Really.  3.x will not work"
+    depends_on('python@:2.9.99')  # docs say: "Really.  3.x will not work"
     depends_on('cmake@2.8.10:')
     depends_on('git')
     depends_on('r+X')
     depends_on('openssl')
 
-    def install(self, spec, prefix):
-        with working_dir('spack-build', create=True):
-            bash = which('bash')
-            cmake('..',
-                  "-DPYTHON=%s" % join_path(spec['python'].prefix.bin,
-                                            'python'),
-                  *std_cmake_args)
-            make()
-            mkdirp('../build')
-            bash('-c', 'cd .. && ./bin/npm --python="$PYTHON" install')
-            bash('-c', 'cd .. && ./bin/node ./ext/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js --python="$PYTHON" rebuild')  # noqa: E501
-            make("install")
+    def cmake_args(self):
+        spec = self.spec
+        options = []
+
+        options.extend([
+            "-DPYTHON=%s" % join_path(spec['python'].prefix.bin, 'python'),
+        ])
+
+        return options
+
+    @run_after('build')
+    def build_node(self):  # or whatever you want to call it
+        bash = which('bash')
+        mkdirp('../build')
```

Comment:
Well, [the installation instructions](https://github.com/rstudio/shiny-server/wiki/Building-Shiny-Server-from-Source) have you create a `tmp` directory, cd into it and do the cmake there.  In that case the `../build` is at the level of the unpacked tarball.  It seems to work either way, but I suspect that since I have write privileges throughout the spack tree I could be scribbling in an inappropriate place and never know.

From first principles, I believe that you are right and it should just be `build` (which works, empirically).

Can you suggest a way for me to test further?